### ISSUE TITLE
Container Engine Refactor (CRI)

### ISSUE THREAD

**Issue by incertum** [falcosecurity/libs#1589]
Continue improvement suggestions from https://github.com/falcosecurity/libs/issues/1550

Prerequisites: More unit and e2e tests, especially also around cgroups.

Details to be added later.

Related:
- https://github.com/falcosecurity/libs/issues/1298
- https://github.com/falcosecurity/libs/pull/1228
- https://github.com/falcosecurity/falco/issues/2708
- https://github.com/falcosecurity/libs/issues/1115

@leogr @Andreagit97 @gnosek

**Comment by incertum** [comment]
/milestone 0.15.0

/assign

but it won't be just one PR or one person working on it, a collaborative effort is needed

**Comment by FedeDP** [comment]
The effort will continue during the next release.
/milestone 0.17.0

**Comment by FedeDP** [comment]
The effort is still going on right? @incertum 
/milestone 0.18.0

**Comment by incertum** [comment]
Yes. So far we cleaned the API up, but we still need to improve the robustness and self-healing mechanisms if possible. I'll post more starting in 2 weeks ...

**Comment by FedeDP** [comment]
Considering https://github.com/falcosecurity/falco/issues/3403, i think we can close this one. (i mean, no big effort/refactor/cleanup will be made on that area of sinsp since we are moving away from it).

/close
